core/src/main/java/io/undertow/server/handlers/resource/ResourceHandler.java

```diff
@@ -458,7 +455,19 @@ public ResourceHandler setContentEncodedResourceManager(ContentEncodedResourceMa
         return this;
     }
 
-    public boolean isCanonicalizePaths() {
+	public HttpHandler getUnsupportedMethodHandler() {
+		return unsupportedMethodHandler;
+	}
+
+	public void setUnsupportedMethodHandler(HttpHandler unsupportedMethodHandler) {
```

Comment:
I know that this does not hold true for some of the older handlers, but in general we prefer handlers that are not routing related to be immutable.

Reply:
What do you mean? You're referring to the "setter" ? Do you suggest this should be done via the constructor ?